api/server/server.go

```diff
@@ -289,9 +291,40 @@ func makeHTTPHandler(logging bool, localMethod string, localRoute string, handle
 
 		w.Header().Set("Server", "Docker/"+dockerversion.VERSION+" ("+runtime.GOOS+")")
 
-		if err := handlerFunc(version, w, r, mux.Vars(r)); err != nil {
-			logrus.Errorf("Handler for %s %s returned error: %s", localMethod, localRoute, err)
-			httpError(w, err)
+		if len(authZPlugins) > 0 {
+
+			// User and UserAuthNMethod are taken from AuthN plugins
+			// Currently tracked in https://github.com/docker/docker/pull/13994
+			user := ""
+			userAuthNMethod := ""
+			ctx := authorization.NewCtx(authZPlugins, user, userAuthNMethod, r.Method, r.RequestURI)
+
+			if err := ctx.AuthZRequest(w, r); err != nil {
+				logrus.Errorf("AuthZRequest for %s %s returned error: %s", localMethod, localRoute, err)
+				httpError(w, err)
+				return
+			}
+
+			rw := authorization.NewResponseModifier(w)
+
+			if err := handlerFunc(version, rw, r, mux.Vars(r)); err != nil {
+				logrus.Errorf("Handler for %s %s returned error: %s", localMethod, localRoute, err)
+				httpError(w, err)
+				return
+			}
+
+			if err := ctx.AuthZResponse(version, rw, r, mux.Vars(r)); err != nil {
```

Comment:
I'm having issues finding a use case for this `AuthZResponse`. Why would we want to let plugins to modify the api response after granting access in the request?